### PR TITLE
added shields for bridging

### PR DESCRIPTION
**Pull request by kevinunger**
- issue #1080 pr #1081 

-Shields are now shown all 3 tabs 
image
image
-Updated the third tab to be more responsive
image
image

- added isBridge prop in NftList.vue that disables filter options

- added css clamps for mobile

-To Do (couldn't find a quick way to split the string for nfts != weapons||chars without breaking stuff; maybe its too late)
https://github.com/CryptoBlades/cryptoblades/blob/37e581d5281daeacf2b01685903f5bfd17dd4f55/frontend/src/views/Bridge.vue#L446-L453
(split the <type>.<id> to --> <id> then do the checks)

### CODE REVIEW

File: frontend/src/views/Bridge.vue [hunk 1]
```diff
@@ -29,6 +29,12 @@
               {{$t('bridge.showCharacters')}}
             </b-button>
           </div>
+          <div class="p-2">
+            <b-button variant="primary" @click="nftType = 'shield'; selectedNftId = ''" class="gtag-link-others"
+              tagname="show_shields_bridge" :disabled="nftType === 'shield'">
+              Show Shields
```
Comment (Chriserus) [inline frontend/src/views/Bridge.vue:35]: Missing translation

File: frontend/src/views/Bridge.vue [hunk 2]
```diff
@@ -48,6 +54,13 @@
         <div class="sell-grid" v-if="nftType === 'character'">
           <character-list :showFilters="true" v-model="selectedNftId" />
         </div>
+
+        <div class="sell-grid" v-if="nftType === 'shield'">
+          <nft-list v-model="selectedNftId"
```
Comment (Chriserus) [inline frontend/src/views/Bridge.vue:59]: Formatting, can be one line

File: frontend/src/views/Bridge.vue [hunk 3]
```diff
@@ -74,9 +87,13 @@
             </b-button>
           </div>
           <div class="p-2">
-            <b-button :disabled="selectedNftId === '' ||
-            (currentTransferNFTId === selectedNftId && transferStatus !== transferStates.restored) ||
-            !storedNftsIds.includes(String(selectedNftId))"
+            <b-button variant="primary" @click="nftType = 'shield'; selectedNftId = ''; getStoredIds()"
+              class="gtag-link-others" tagname="show_shield_bridge" :disabled="nftType === 'shield'">
+              Show Shields
```
Comment (Chriserus) [inline frontend/src/views/Bridge.vue:92]: Missing translation

File: frontend/src/views/Bridge.vue [hunk 4]
```diff
@@ -74,9 +87,13 @@
             </b-button>
           </div>
           <div class="p-2">
-            <b-button :disabled="selectedNftId === '' ||
-            (currentTransferNFTId === selectedNftId && transferStatus !== transferStates.restored) ||
-            !storedNftsIds.includes(String(selectedNftId))"
+            <b-button variant="primary" @click="nftType = 'shield'; selectedNftId = ''; getStoredIds()"
```
Comment (Chriserus) [inline frontend/src/views/Bridge.vue:90]: I would extract the @click to a method, as it has 3 actions going on here

File: frontend/src/views/Bridge.vue [hunk 5]
```diff
@@ -165,6 +190,20 @@
         <div v-else-if="nftType === 'character'">
           <h3 class="text-center p-4">{{$t('bridge.noCharactersStored')}}</h3>
         </div>
+        <div v-if="nftType === 'shield' && storedNftsIds.length !== 0">
+          <nft-list
+          v-model="selectedNftId"
+          :characterIds="[storedNftsIds]"
```
Comment (Chriserus) [inline frontend/src/views/Bridge.vue:196]: Can you please check again if we need all of these attributes? We're showing shields here, but we're including :characterIds and we are using nftDysplay flag, is this needed?

File: frontend/src/views/Bridge.vue [hunk 6]
```diff
@@ -210,12 +249,29 @@
...
               </select>
-              <div class="p-2 text-center">
+              <div class="mt-2 text-center">
                 <b-button :disabled="!characterIdToWithdraw" variant="primary"
-                  @click="withdrawBridge(characterIdToWithdraw)" class="gtag-link-others"
+                  @click="withdrawBridge(characterIdToWithdraw)" class="gtag-link-others withdrawBtn"
                   tagname="click_transfer_bridge">{{$t('bridge.withdrawCharacter')}}</b-button>
               </div>
             </div>
+            <div v-if="incomingShields.length !== 0" class="p-4 w-20">
```
Comment (Chriserus) [inline frontend/src/views/Bridge.vue:258]: Missing translations in this div

File: frontend/src/views/Bridge.vue [hunk 7]
```diff
@@ -193,15 +232,15 @@
...
               </div>
             </div>
             <div v-if="incomingChars.length !== 0" class="p-4 w-20">
-              <h4 class="text-center">{{$t('bridge.selectCharacterToWithdraw')}}</h4>
-              <select class="form-control" v-model="characterIdToWithdraw">
+              <h4 class="text-center withdrawText">{{$t('bridge.selectCharacterToWithdraw')}}</h4>
+              <select class="form-control withdrawSelect" v-model="characterIdToWithdraw">
                 <option value="" disabled selected>{{$t('bridge.selectYourCharacter')}}</option>
                 <option v-for="character in incomingChars"
                 :value="character['7']"
```
Comment (Chriserus) [inline frontend/src/views/Bridge.vue:246]: If there was enough time for that, I would refactor the way we are handling types in a bridge, character should be a type here, we should access a particular attribute, not access it via '7', similar in other places in code here

File: frontend/src/views/Bridge.vue [hunk 8]
```diff
@@ -622,4 +708,14 @@ export default Vue.extend({
...
 	transform: scale(1);
 }
+.withdrawText{
+  font-size: clamp(12px, 1.25rem, 24px);
+}
+.withdrawSelect{
+  width: clamp(200px, 20vw, 300px);
+  margin:0 auto;
+}
+.withdrawBtn{
```
Comment (Chriserus) [inline frontend/src/views/Bridge.vue:718]: If you want extra-clean css, you can add it like that:
.withdrawSelect{
  margin:0 auto;
}

.withdrawSelect,
.withdrawBtn{
  width:clamp(200px, 20vw , 300px);
}

File: frontend/src/views/Bridge.vue [hunk 9]
```diff
@@ -451,6 +524,9 @@ export default Vue.extend({
       return fromWeiEther(wei);
     },
     async transferToStorage(){
+      // chars & weapon ids are ints; other nfts are <type.id>
+      if(this.nftType !== 'weapon' && this.nftType !== 'character') this.selectedNftId = this.selectedNftId.split('.')[1];
```
Comment (Chriserus) [inline frontend/src/views/Bridge.vue:528]: I'll look into it a bit, but maybe it's possible to map the ids if they are different? Also, that's bridge-specific, right? Bcs I'm using fetching weapons/junk/shields and all that by ids and it's all done in a similar manner, not some kind of type.id. I'll look into it and come back to you

**Review by Chriserus — CHANGES_REQUESTED** [review]

File: frontend/src/views/Bridge.vue (hunk 4 above)
Comment (Chriserus) [reply frontend/src/views/Bridge.vue:90]: lol, I tagged some guy by accident
image

If you're reading it, cheers mate :)

File: frontend/src/views/Bridge.vue (hunk 1 above)
Comment (kevinunger) [reply frontend/src/views/Bridge.vue]: resolved in 921a343f093d493a8de431cf85e0ba426d0eeb73

File: frontend/src/views/Bridge.vue (hunk 6 above)
Comment (kevinunger) [reply frontend/src/views/Bridge.vue:257]: resolved in 9153fdf622c7d8f6f35ae769c2caa66d20392fda

**Pull request merged**